src/qiskit_qasm3_import/expression.py

```diff
@@ -233,6 +241,13 @@ def visit_BinaryExpression(self, node: ast.BinaryExpression):
                 elif isinstance(rhs_type, types.Angle):
                     const = lhs_type.const and rhs_type.const
                     out_type = types.Uint(const, None)
+                # We allow `angle / float` in non-strict mode, because the Qiskit OQ3 exporter does
+                # not / cannot handle `ParameterExpression` well on output, and often outputs things
+                # like that.  That's invalid OQ3, but it's better to support Qiskit's dodgy output
```

Comment:
Is there an issue tracking this in the language specification? Or was this already discussed and decided that this should not be allowed. It seems odd to me though if this was the decision since it seems pretty natural to perform arithmetic operations between an angle and a float.

Reply:
An `angle` is an integer-like representation as fractions of $2\pi$, and was more intended for exact calculations and for close-to-QPU classical hardware that doesn't have floating-point calculations at runtime, so when it's been previously discussed, it was deliberate that `float` doesn't interact with `angle`.  That said, there's still awkwardnesses in the whole model around the `angle` type being naturally defined on a ring.